docs/mkdocs/docs/api/macros/json_diagnostic_positions.md

```diff
@@ -6,22 +6,36 @@
 
 This macro enables position diagnostics for generated JSON objects.
 
-When enabled, two new properties: `start_pos()` and `end_pos()` are added to `nlohmann::basic_json` objects and fields. `start_pos()` returns the start 
-position of that JSON object/field in the original string the object was parsed from. Likewise, `end_pos()` returns the end position of that JSON
-object/field in the original string the object was parsed from.
-
-`start_pos()` returns the first character of a given element in the original JSON string, while `end_pos()` returns the character following the last
-character. For objects and arrays, the first and last characters correspond to the opening or closing braces/brackets, respectively. For fields, the first
-and last character represent the opening and closing quotes or the first and last character of the field's numerical or predefined value
-(true/false/null), respectively.
-
-Given the above, `end_pos() - start_pos()` for an object or field provides the length of the string representation for that object or field, including the
-opening or closing braces, brackets, or quotes.
-
-`start_pos()` and `end_pos()` are only set if the JSON object was parsed using `parse()`. For all other cases, `std::string::npos` will be returned.
-
-Note that enabling this macro increases the size of every JSON value by two `std::size_t` fields and adds
-slight runtime overhead.
+When enabled, two new member functions [`start_pos()`](../basic_json/start_pos.md) and
+[`end_pos()`](../basic_json/end_pos.md) are added to [`basic_json`](../basic_json/index.md) values.
+[`start_pos()`](../basic_json/start_pos.md) returns the start position of that JSON value in the original JSON string
+the object was parsed from. Likewise, [`end_pos()`](../basic_json/end_pos.md) returns the end position of that JSON
+value in the original string the object was parsed from.
```

Comment:
If the JSON value was not parsed from text but was instead constructed directly, these functions will return `std::string::npos`.

Update: I see that's covered below.  Don't need to duplicate it.  Maybe the section below should be moved here.

Reply:
Done in 37bb4d9c5d504ff0e8a9cd00cb33c84009ebcf95